usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ValidateCalendarEndDateBeforeStartDateNoticeTest.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2020. MobilityData IO.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.usecase;
+
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.Test;
+import org.mobilitydata.gtfsvalidator.domain.entity.gtfs.Calendar;
+import org.mobilitydata.gtfsvalidator.domain.entity.notice.error.CalendarEndDateBeforeStartDateNotice;
+import org.mobilitydata.gtfsvalidator.usecase.port.GtfsDataRepository;
+import org.mobilitydata.gtfsvalidator.usecase.port.ValidationResultRepository;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+/**
+ * Unit tests to validate behavior of use case for "E032 - calendar.txt end_date is before start_date"
+ */
+class ValidateCalendarEndDateBeforeStartDateNoticeTest {
+
+    // TODO - below test throws an NPE, but calendar.txt start_date and end_date are required and marked @NonNull,
```

Comment:
Flagging this as a question for discussion

Reply:
IMO, we manipulate `Calendar` entities in the use case, given that these entities are only built if `non-null` fields are actually `non-null`; if I am correct, we could skip null tests for required GTFS fields.
However, we should include them for `optional` GTFS fields.

Reply:
I agree with @lionel-nj , we have builders that shield us from having nulls downstream, hence the NPE is to be expected.

Reply:
Sounds good! I just removed this commented-out test in https://github.com/MobilityData/gtfs-validator/pull/272/commits/8925240802796cb7ed6b315e82196279af7dd3ac.